### DISCUSSION TITLE
Automatically logout

### DISCUSSION THREAD

**Discussion by Nerd-Ecke** [authelia/authelia#8010] [Q&A]
Hy

I would like the session to log out automatically and continue to the login screen when I log in somewhere else.

I have configured redis and set maximum_active_connections: 1 

But the other session is not logged out until the idle time has expired, but is not redirected to the loginscreen until I refresh the tap.

Is this behaviour even possible with Authelia?

Use nginx as proxy.

I am grateful for all solutions.

**Comment by Crowley723** [comment]
> I would like the session to log out automatically and continue to the login screen when I log in somewhere else.

You want authelia to only have a single logged in session for your user?
What's the benefit of this over setting an idle?
This seems like it might be too much of a hassle for users to have to login to every session every time you switch computers.

> But the other session is not logged out until the idle time has expired but is not redirected to the loginscreen until I refresh the tap.

Once the session expires, no access is allowed. As soon as a new request with the expired cookie is sent, your browser should be redirected to authelia. 

As for the feasibility, currently, the session library authelia uses doesn't directly support session management, so it's really difficult to even retrieve a list of a single user's sessions. This may be more feasible with the planned session rework, but that won't come until v4.40 at the very earliest.

**Reply by Nerd-Ecke** [reply]
> You want authelia to only have a single logged in session for your user? What's the benefit of this over setting an idle? This seems like it might be too much of a hassle for users to have to login to every session every time you switch computers.
> 
Yes, I would like Authelia to have the option of having only a single login session. 
The advantage over setting an idle session is for security critical applications. e.g. password manager.

> As for the feasibility, currently, the session library authelia uses doesn't directly support session management, so it's really difficult to even retrieve a list of a single user's sessions. This may be more feasible with the planned session rework, but that won't come until v4.40 at the very earliest.

Then I am looking forward to v4.40.

Thanks for the feedback.

**Reply by Crowley723** [reply]
> The advantage over setting an idle session is for security critical applications. e.g. password manager.

What advantage are you thinking of? Most password manager's encrypt your vault when not in use and you have to give your Master Password to decrypt it. I fail to see how having a single session greatly increases the security of the application.

@james-d-elliott Brought up a good point. Having only a single session you can login to opens you up to a denial of service attack if your Authelia password gets compromised.

Its unlikely we will support limiting the number of user sessions.

**Comment by james-d-elliott** [comment]
As discussed above this will be unlikely to be supported due to security issues with the particular implementation.
